status/condition.go

```diff
@@ -26,6 +26,10 @@ const (
 	ConditionSucceeded = "Succeeded"
 )
 
+// ConditionDegraded is a custom extension to the standard Kubernetes condition statuses.
+// It represents a state where the resource is operational but in a degraded state.
+const ConditionDegraded metav1.ConditionStatus = "Degraded"
```

Comment:
I'd be wary of supporting concepts that aren't known concepts in upstream K8s. 

Why is this not the same thing as "Unknown"? That's sort of the implication when you don't have the info in the same way as Degraded.